PanoptesNetClient/PanoptesNetClient/ApiClient.cs

```diff
@@ -32,23 +36,125 @@ public ApiClient(HttpClient client = null)
             }
         }
 
-        public async Task<JObject> GetAsync(IRequest request)
+        /// <summary>
+        /// Make a GET request for a single resource using IRequest
+        /// </summary>
+        #region Generic GET using IRequest
+        public async Task<T> Get<T>(IRequest request)
         {
-            JObject resource = null;
+            List<T> collection = await GetAsync<T>(request.Endpoint);
 
-            HttpResponseMessage response = await Client.GetAsync(request.Endpoint);
+            if (collection.Count > 0)
+            {
+                return collection[0];
+            }
+            return default(T);
+        }
+        #endregion
+
+        /// <summary>
+        /// Make a GET request for a single resource using an endpoint
+        /// </summary>
+        #region Generic GET using string
+        public async Task<T> Get<T>(string route)
+        {
+            var att = (Path)Attribute.GetCustomAttribute(typeof(T), typeof(Path));
```

Comment:
I'm not a huge fan of this line of code because there is no way to guarantee that the class passed in will have a custom attribute of type `Path`. I'm not really sure if it's a big deal, but would be nice if there was a way to avoid this.

Reply:
Also, whether you change to code here or not, can you change the name from `att` to `path` or something?

Reply:
I can remove this, but it will change the structure of the code a good deal. Many C# rest clients I've seen have a different client (ProjectClient, SubjectClient) that inherit from a base client to avoid this sort of generalization. 

Yes, this is a bit of a code smell, I'll admit, and it was mainly done to avoid having to create separate client instances. Lets chat about this next week. I'm happy to make the change if we think it will make a more robust client.